### PR TITLE
fix: don't debounce if component receives the same props

### PR DESCRIPTION
**Pull request by nikhedonia**
https://www.uat-thetimes.co.uk/profile/giles-coren seems to put the component into debounce mode right after mount even though the data is already in apollos inmemory store.

This PR adds a deep comparison of props and nextProps and sets the isLoading prop once the component stopped debouncing.

### CODE REVIEW

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/4ce44d577dd177304e717865024b9fb8e51285bc/dextrose-test/index.html

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/7b47681e272363500047ff8b545175f6d30529b1/dextrose-test/index.html

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/51fdb0301a813a0328b84b103d77f4c82cc1eabd/dextrose-test/index.html

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/57305284ca1ad51453feadecbc082e0bce242b96/dextrose-test/index.html

File: packages/provider/__tests__/provider-state-transition.test.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    author(slug: $slug) {
+      name
+    }
+  }
+`;
+
+const Connected = connect(query);
+const Debounced = props => <Connected {...props} debounceTimeMs={1000} />;
+
+const wait = ms => {
```
Comment (craigbilner) [inline packages/provider/__tests__/provider-state-transition.test.js]: we should move into `utils` now

Comment (nikhedonia) [reply packages/provider/__tests__/provider-state-transition.test.js]: seperate pr maybe?

**Pull request merged**